pkg/environments/implementation.go

```diff
@@ -81,11 +85,22 @@ func (env *environment) WaitForReady(ctx context.Context) chan error {
 	errs := make(chan error)
 
 	go func() {
+		// if the cluster fails to become ready after N minutes, assume it's likely stuck and dump a diagnostic bundle.
+		// this uses its own timer since we can't catch "go test" timeouts via the ctx.
+		hung := time.AfterFunc(readyHungDuration, func() {
```

Comment:
We might want to check if context has deadline via `ctx.Deadline()` and if it does use it here. Otherwise fallback to a const.

This can be handy in tests using `t.Deadline()` which could then be passed to context to `WaitForReady()`

WDYT?

Reply:
If we remember to use it, yes.

I think we just collect on any done condition then, deadline or otherwise, since it expiring here should mean _something_ unusual happened. Will maybe generate diagnostics if you terminate the process here, but whatever, tmpfiles are cheap.